### ISSUE TITLE
clr-modal does not have aria-labelledby

### ISSUE THREAD

**Issue by zivanov123** [vmware/clarity#2746]
## Describe the bug

Modals and wizards do not have associated aria-labelledby.
This is required so accessibility software to be able to announce the dialogs correctly.
Currently modals have only role='dialog'.

More info can be found here:
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/dialog_role

## How to reproduce
Can be seen on the Clarity site.

## Expected behavior

The div element with role='dialog' to have aria-labelledby. Probably 'modal-header' can get an id and be used.

## Versions

**App**

* Angular: [6]
* Node: [8.10.0]
* Clarity: [0.11.32]

**Device:**

* Type: [any]
* OS: [any]
* Browser [any]
* Version [any]

**Comment by mathisscott** [comment]
I tried but verified this is too much to port back to 0.11. It will go into the next version of 0.12 tomorrow.
